easybuild/framework/easyconfig/types.py

```diff
@@ -126,18 +126,53 @@ def to_toolchain(tcspec):
         if len(tcspec) == 2:
             res = {'name': tcspec[0].strip(), 'version': tcspec[1].strip()}
         else:
-            raise EasyBuildError("Can not convert list %s to toolchain dict. Expected 2 elements", tcspec)
+            errstr = "Can not convert list %s to name and version dict. Expected 2 elements"
+            raise EasyBuildError(errstr, tcspec)
     else:
-        raise EasyBuildError("Conversion of %s (type %s) to toolchain dict is not supported", tcspec, type(tcspec))
+        errstr = "Conversion of %s (type %s) to name and version dict is not supported"
+        raise EasyBuildError(errstr, tcspec, type(tcspec))
 
     return res
 
 
-# this uses to_toolchain, so it needs to be at the bottom of the module
+def to_dependency(dep):
+    """
+    Convert a dependency dict from the new YAML format {'lib': '1.2.8', 'toolchain': 'gcc, 4.8.2'}
+    to a dependency dictionary
+    """
+    depspec = {}
+    if isinstance(dep, dict):
+        for key, value in dep.items():
+            if key == 'versionsuffix':
+                depspec['versionsuffix'] = value
+            elif key == 'toolchain':
+                depspec['toolchain'] = to_name_version_dict(value)
+            elif 'name' not in depspec:
+                depspec.update({'name': key, 'version': value})
+            else:
+                raise EasyBuildError("Found unexpected key, value pair: %s, %s", key, value)
+
+        if 'name' not in depspec:
+            raise EasyBuildError("Can not parse dependency without name and version: %s", dep)
+
+    else:
+        raise EasyBuildError("Can not convert %s (type %s) to dependency dict", dep, type(dep))
+
+    return depspec
+
+
+def to_dependencies(dep_spec_list):
+    """
+    Convert a list of dependencies in yeb format to a list of dependency tuples
+    """
+    return [to_dependency(dep_spec) for dep_spec in dep_spec_list]
+
+
+# this uses functions defined in this module, so it needs to be at the bottom of the module
 TYPE_CONVERSION_FUNCTIONS = {
     basestring: str,
     float: float,
     int: int,
     str: str,
-    dict: to_toolchain,
+    dict: to_name_version_dict,
```

Comment:
hmm, this implies that the auto-conversion will try to convert everything that should be a `dict` but is not using `to_name_version_dict`, which is probably not what we want in the long term... we need a custom type `name_version_dict`?